test/cqlpy/test_secondary_index.py

```diff
@@ -1970,3 +1971,21 @@ def test_index_in_system_tables(cql, test_keyspace):
         assert f'{test_keyspace}::{index_name}' in res
         res = cql.execute(f'select * from system."IndexInfo" where table_name = \'{test_keyspace}\' AND index_name = \'{index_name}\'').one()
         assert (test_keyspace, index_name) == (res.table_name, res.index_name)
+
+# Test index representation in API
+def test_index_in_API(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int") as table:
+        index_name = unique_name()
+        cql.execute(f"CREATE INDEX {index_name} ON {table}(v)")
+        # Index only appears in the API report after being built, so give it some time
+        # Calling wait_for_index() is not enough. The call waits for index to appear in the
+        # system.built_views table, while the API check system.scylla_views_builds_in_progress
```

Comment:
Wow, I didn't expect this. It looks like a bug in the API that we'll need to fix and not just enshrine the bug in the test. Isn't the build_views the **official** tracker of which views completed building, and the views_builds_in_progress the less important one?

Reply:
Created https://github.com/scylladb/scylladb/issues/21587

Reply:
Cassandra's "official" way of determining when an index build has finished (remember that Cassandra's SI is not a view) is the `system."IndexInfo"` table. So this is what the wait_for_index() function in the test uses.
Our implementation that is index/built_indexes_virtual_reader.hh and uses built_views, not view_build_status...
What a mess :-(